src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -88,6 +88,10 @@
 
     private static final String CONTENT_TYPE_JSON = "application/json";
 
+    // Android cannot hold unlimited changes data.
+    // See ChangesOptions.limit default value (Integer.MAX_VALUE)
+    public static final int DEF_CHANGES_LIMIT = 10000;
```

Comment:
Does this have a side effect to one-short pull replicator for P2P? I don't know if there is a better fix or not, but each device / platform has different memory limitation. Just want to know if the users do this by themselves?

Reply:
@pasin Thanks for pointing about the oneshot pull replication peer-to-peer scenario. I will test one shot and continuous pull replication.